### PR TITLE
Ease Federated App deployment on Workload Plane

### PR DESCRIPTION
**Pull request by MonPote**
**Component**:

salt, csc

**Context**: 

In order to deploy federated app in an easier way on the workloadplane, we introduce here a new workloadplane configuation for Shell UI and a new default backend for the WP ingress.

### CODE REVIEW

File: tests/post/steps/test_ingress.py [hunk 1]
```diff
@@ -569,6 +569,24 @@ def server_returns(host, context, status_code, reason):
     assert response.status_code == int(status_code)
     assert response.reason == reason
 
+@then(
+    parsers.re(r"the server should respond with shell-ui index"),
+    converters=dict(status_code=int),
```
Comment (TeddyAndrieux) [inline tests/post/steps/test_ingress.py]: It's not needed here

File: tests/post/steps/test_ingress.py [hunk 2]
```diff
@@ -569,6 +569,24 @@ def server_returns(host, context, status_code, reason):
...
+)
+def shell_ui_returns(host, context):
+    response = context.get("response")
+    assert response is not None
+    assert "window.shellUIRemoteEntryUrl = \"/shell/remoteEntry.js" in response.text
+
+@then(
+    parsers.re(r"the server should not respond with shell-ui index"),
+    converters=dict(status_code=int),
```
Comment (TeddyAndrieux) [inline tests/post/steps/test_ingress.py]: ditto

File: tests/post/steps/test_ingress.py [hunk 3]
```diff
@@ -569,6 +569,22 @@ def server_returns(host, context, status_code, reason):
     assert response.status_code == int(status_code)
     assert response.reason == reason
 
+@then(
+    parsers.re(r"the server should respond with shell-ui index"),
+)
```
Comment (TeddyAndrieux) [inline tests/post/steps/test_ingress.py]: nit:

```suggestion
@then("the server should respond with shell-ui index")
```

File: salt/metalk8s/addons/nginx-ingress/deployed/init.sls [hunk 4]
```diff
@@ -6,3 +6,34 @@ include:
...
+    - apiVersion: v1
+    - kind: Service
+    - name: ingress-nginx-defaultbackend
+    - namespace: metalk8s-ingress
+    - require:
+      - sls: metalk8s.addons.nginx-ingress.deployed.chart
+
+Delete old deployment for the default backend:
+  metalk8s_kubernetes.object_absent:
+    - apiVersion: v1
```
Comment (TeddyAndrieux) [inline salt/metalk8s/addons/nginx-ingress/deployed/init.sls]:
```suggestion
    - apiVersion: apps/v1
```

**Review by JBWatenbergScality — APPROVED** [review]

**Comment by MonPote** [comment]
ping

**Pull request merged**